CMakeLists.txt

```diff
@@ -287,15 +287,18 @@ if (ADDRESS_SANITIZER_DEBUG)
 endif()
 
 
-# Note: it is better to include `eigen3` `async` as
+# Note: it is better to include `async` as
 # system headers because they emit lot's of warnings
 # from clang. Most of them are issues with respect
 # to overriden virtual methods
 target_include_directories(SeisSol-lib SYSTEM PUBLIC
-  ${CMAKE_CURRENT_SOURCE_DIR}/submodules/eigen3
   ${CMAKE_CURRENT_SOURCE_DIR}/submodules/async
 )
 
+
+find_package(Eigen3 3.4.0 REQUIRED)
+target_include_directories(SeisSol-lib SYSTEM PUBLIC ${EIGEN3_INCLUDE_DIRS})
```

Comment:
Wouldn't it be better to do it like this:
https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html
`target_link_libraries (SeisSol-lib PUBLIC Eigen3::Eigen)`